src/log/log_private.h

```diff
@@ -312,42 +277,6 @@ struct __wt_log {
     uint32_t flags;
 };
 
-struct __wt_log_record {
-    uint32_t len;      /* 00-03: Record length including hdr */
-    uint32_t checksum; /* 04-07: Checksum of the record */
-
-/*
- * No automatic generation: flag values cannot change, they're written to disk.
- *
- * Unused bits in the flags, as well as the 'unused' padding, are expected to be zeroed; we check
- * that to help detect file corruption.
- */
-#define WT_LOG_RECORD_COMPRESSED 0x01u /* Compressed except hdr */
-#define WT_LOG_RECORD_ENCRYPTED 0x02u  /* Encrypted except hdr */
-#define WT_LOG_RECORD_ALL_FLAGS (WT_LOG_RECORD_COMPRESSED | WT_LOG_RECORD_ENCRYPTED)
-    uint16_t flags;    /* 08-09: Flags */
-    uint8_t unused[2]; /* 10-11: Padding */
-    uint32_t mem_len;  /* 12-15: Uncompressed len if needed */
-    uint8_t record[0]; /* Beginning of actual data */
-};
-
-/*
- * __wt_log_record_byteswap --
- *     Handle big- and little-endian transformation of the log record header block.
- */
-static WT_INLINE void
-__wt_log_record_byteswap(WT_LOG_RECORD *record)
-{
-#ifdef WORDS_BIGENDIAN
-    record->len = __wt_bswap32(record->len);
-    record->checksum = __wt_bswap32(record->checksum);
-    record->flags = __wt_bswap16(record->flags);
-    record->mem_len = __wt_bswap32(record->mem_len);
-#else
-    WT_UNUSED(record);
-#endif
-}
-
 /*
  * WT_LOG_DESC --
```

Comment:
currently __wt_log_desc is being used in [tools/hexfiend/Templates/WT/WT-Log.tcl](https://github.com/wiredtiger/wiredtiger/blob/ba14e16959924a2a970998878e104a04bb79587b/tools/hexfiend/Templates/WT/WT-Log.tcl#L35)

I am not sure about the usage of this file, if it is required, may be we need keep it in log.h

Reply:
I ran hexfiend with the WT_Log template to confirm, and there's no issues here. 
Attached below is a screenshot of hexfiend decoding the LOG_DESC field correctly (see the blue highlighted section on the right). 
<img width="1741" alt="Screenshot 2024-10-14 at 10 36 20 AM" src="https://github.com/user-attachments/assets/45ae7ba4-9153-4920-ac16-d43ee8cca0bf">

Hexfiend templates are an independent definition of the on-disk data layout. This should be stable given it's on disk, but it's a good callout that hexfiend has a dependency on on-disk data formatting.